app/controllers/provider/admin/account/data_exports_controller.rb

```diff
@@ -32,21 +45,32 @@ def authorize_data_export
     authorize! :export, :data
   end
 
-  def redirect_to_new_on_invalid_data
-    #TODO: test this behaviour
-    redirect_to :action => :new unless EXPORTS_TARGETS.include?(permitted_params[:data])
+  def set_selects_collections
+    @targets = permitted_types.map(&:reverse)
+    @periods = PERIODS.map(&:reverse)
   end
 
-  def set_selects_collections
-    @targets = EXPORTS_LABELS.except(current_account.settings.finance.allowed? ? nil : :invoices).map(&:reverse)
+  def valid_params?
+    if permitted_types.keys.exclude?(data.to_sym)
+      flash.now[:danger] = t('.invalid_data')
+      false
+    elsif PERIODS.keys.exclude?(period.to_sym)
+      flash.now[:danger] = t('.invalid_period')
+      false
+    else
+      true
+    end
+  end
 
-    @periods = [['All', ''], ['Today', 'today'], ['This Week', 'this_week'],
-                ['This Month', 'this_month'], ['This Year', 'this_year'],
-                ['Previous Year', 'last_year']]
+  def data
+    @data ||= params.require(:export).require(:data)
   end
 
-  def permitted_params
-    params.permit(:data, :period)
+  def period
+    @period ||= params.require(:export).require(:period)
   end
```

Comment:
```suggestion
  def data
    export_params[:data]
  end

  def period
    export_params[:period]
  end
  
  def export_params
    @export_params ||= params.require(:export)
  end
```

I guess I'd do smth like that to process empty params successfully.

Reply:
No strong opinions about this, but

There is no reason why :data or :period would be missing. The only possibility is the user deletes one the selects from then HTML.

When using `require` the server returns:
```
Required parameter missing: period
```

Using `&.` in my opinion is misleading (because it gives the impression it is an optional parameter).

IMO the best would be `expect`, but it was introduced in rails 8.